add LICENSE file

etienne-lms opened the pull request:
Add LICENSE file with the BSD-2-Clause terms.

Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>

etienne-lms committed:
add LICENSE file

Add LICENSE file with libyaml terms for libyaml files and
BSD-2-Clause terms for other files.

Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>
Acked-by: Jerome Forissier <jerome.forissier@linaro.org>

jforissier commented on LICENSE:
```diff
@@ -0,0 +1,27 @@
+Unless it has its own copyright/license embedded in its body, each source file
```
`s/source//` to cater for documentation files etc.

But, there is a more fundamental problem with this statement: some files in `libyaml/` do not have any copyright/license statement and they are neither copyrighted by Linaro nor covered by BSD 2-Clause...

etienne-lms replied:
Do we copyright the docs? especially since docs have moved to `optee_docs`?
Yet, ok to remove reference to "source". Maybe this change could also replace the license notice from source and header files with a SPDX license tag.

Regarding yaml, libyaml is release under MIT license terms. Maybe both license terms (MIT and BSD-2-Clause)  should be listed in this `LICENSE` file.

etienne-lms committed:
Replace inline license terms with SPDX license tags

This change covers the source files of the benchmark embedded tool
main functions. Yaml library is not modified, as is it better
synchronizes with the generic libyaml source and version trees.

Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>
Acked-by: Jerome Forissier <jerome.forissier@linaro.org>

etienne-lms commented:
ping!
It would be much better that this package (optee_benchmark) comes with a LICENSE file. Many distro packaging expect each package clearly state their license term from such a file.

jforissier commented:
For "Replace inline license terms with SPDX license tags":
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

jforissier commented on LICENSE:
```diff
@@ -1,8 +1,8 @@
-is subject to the following license terms:
+This software contains materials under the terms of the BSD-2-Clause license
+and materials under the terms of the MIT license as stated in the related
+files.
```
"as stated in the related files" is ambiguous, or even incorrect, because libyaml files contain no licensing terms (although libyaml/ has a LICENSE file).

IMO we should be more explicit:

```
1. Files under libyaml/ are covered by the following license:

<Insert libyaml/LICENSE here>

2. All other files are covered by the following license (BSD-2-Clause):

<Insert BSD-2-Clause text here>
```

etienne-lms replied:
ok, will do. thanks.

jforissier commented:
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

etienne-lms commented:
tag applied.

etienne-lms commented:
can we merge this for tag 3.5.0?

jockebech commented:
Absolutely, will merge right away, thanks!

igoropaniuk commented:
@jforissier @jbech-linaro btw, am I still eligible to merge PRs here?

jockebech commented:
@igoropaniuk so far we haven't changed anything, so you can still merge. However, since your accountability might be different than before, it might be that someone else than you also will do the merge.
